miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -320,24 +232,55 @@ private long resolveMembers(SecurityProfile sp) throws IOException {
     List<User> resolved = Lists.newArrayList();
     if (users != null) {
       for (User user : users) {
-        User u = securityStore.getUserById(user.getUserId());
+        User u = securityManager.getUserById(user.getUserId());
         if (u == null) throw new IllegalArgumentException("User " + user.getUserId() + " does not exist");
         resolved.add(u);
       }
     }
     return resolved;
   }
 
-  private Collection<Group> resolveGroups(Collection<Group> groups) throws IOException {
-    List<Group> resolved = Lists.newArrayList();
-    if (groups != null) {
-      for (Group group : groups) {
-        Group g = securityStore.getGroupById(group.getGroupId());
-        if (g == null) throw new IllegalArgumentException("Group " + group.getGroupId() + " does not exist");
-        resolved.add(g);
+  @Override
+  public long saveProject(Project project) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)